storage/innobase/row/row0vers.cc

```diff
@@ -530,16 +528,12 @@ row_vers_build_cur_vrow_low(
 			 = DATA_MISSING;
 	}
 
-	ut_ad(mtr->memo_contains_page_flagged(rec,
-					      MTR_MEMO_PAGE_S_FIX
-					      | MTR_MEMO_PAGE_X_FIX));
-
```

Comment:
It was a mistake to remove this code in c62493628d6175b9a7880856aaaf876c28a9af82. Initially I was removing `mtr`  parameters that had only been used for debug assertions. In a later iteration, I put back the `mtr` parameter to this function so that we could use the same object for the "sub-mini-transactions" that would access undo log pages to fetch the previous version.

Reply:
The same mistake was made also in `row_purge_vc_matches_cluster()`.